### ISSUE TITLE
Attempting to run individual statements or clicking '🔒 Active Connection' results in error messages despite existing connection.

### ISSUE THREAD

**Issue by SebLore** [mtxr/vscode-sqltools#1350]
**Describe the bug**
'No active connection found' and 'You need to create the connection first!' messages for trying to run statements when active connection exists. Can use the command 'Run on active connection' for blocks but not Run on individual statements.

**To Reproduce**
1. Go to SQLTools>Connections>Add New Connection
2. Set up connection to local database, using localhost, port 3306, database name 'user' password 'password' password mode save as plaintext.
3. Test connection, confirm "successfully connected" message. Save connection.
4. Press 'CONNECT NOW'.
5. Write a simple statement like CREATE TABLE test(id INT PRIMARY KEY AUTO_INCREMENT);
6. Click 'Run' above the statement.

**Expected behavior**
The query triggers as expected, creating a new table called test with one empty column.

**Screenshots**
Enter data:
![image](https://github.com/mtxr/vscode-sqltools/assets/34066570/2d6d7acd-bedb-4a54-a9f8-cbec78c94b5b)

Test connection:
![image](https://github.com/mtxr/vscode-sqltools/assets/34066570/bebbb005-8a74-4e31-937e-3b242e80cdea)

Connection exists:
![image](https://github.com/mtxr/vscode-sqltools/assets/34066570/ac52c86f-4764-4725-a983-8b1049c21db7)

'Connect now':
![image](https://github.com/mtxr/vscode-sqltools/assets/34066570/41117726-80a5-409b-8e51-8495411a0042)

Fault occurs:
![image](https://github.com/mtxr/vscode-sqltools/assets/34066570/8ab54e36-74ce-4e0f-8c65-46f196423231)

Possibly relevant warnings:
![image](https://github.com/mtxr/vscode-sqltools/assets/34066570/d6169af3-d045-4c1f-a874-bcecccd0b930)

**Desktop (please complete the following information):**
 - SQLTools Version: v0.28.3
 - VS Code Version: v.1.90.2
 - OS: Windows
 - Driver:
   - [ x ] MySQL/MariaDB
 - Database version: [e.g. MySQL v5.6, PostgreSQL vX...]
 MySQL Server 8.4.
 

**Additional context**
Was following a [tutorial by Fireship](https://www.youtube.com/watch?v=Cz3WcZLRaWc) to set up my database and got stuck on not being able to run code on blocks ( --  @block) and noticed it didn't work.

**Comment by StrangeWill** [comment]
Yeah I'm left clicking "Run on active connection" which kills all hot keys for "Run" -- slightly annoying.

**Comment by MadiBow** [comment]
I also had this problem, but realised I also had the "Database Client" extension installed and this was giving the "Run" option for all .sql files, hence no database connection was found (for my Database Client). The option "Run on active connection" is SQLTools' only run option - but no hotkey seems to exist for this yet, even after removing the Database Client extension.
